source/_components/media_player.kodi.markdown

```diff
@@ -28,20 +28,57 @@ media_player:
     host: 192.168.0.123
 ```
 
-Configuration variables:
-
-- **host** (*Required*): The host name or address of the device that is running XBMC/Kodi.
-- **port** (*Optional*): The HTTP port number. Defaults to 8080.
-- **tcp_port** (*Optional*): The TCP port number. Defaults to 9090. Used for websocket connections to Kodi.
-- **name** (*Optional*): The name of the device used in the frontend.
-- **proxy_ssl** (*Optional*): Connect to kodi with HTTPS and WSS. Defaults to `false`. Useful if Kodi is behind an SSL proxy.
-- **username** (*Optional*): The XBMC/Kodi HTTP username.
-- **password** (*Optional*): The XBMC/Kodi HTTP password.
-- **turn_on_action** (*Optional*): Home Assistant script sequence to call when turning on.
-- **turn_off_action** (*Optional*): Home Assistant script sequence to call when turning off.
-- **enable_websocket** (*Optional*): Enable websocket connections to Kodi via the TCP port. Defaults to `true`. The websocket connection allows Kodi to push updates to Home Assistant and removes the need for Home Assistant to poll. If websockets don't work on your installation this can be set to `false`.
-- **timeout** (*Optional*): Set timeout for connections to Kodi. Defaults to 5 seconds.
-
+{% configuration %}
+host:
+  description: The host name or address of the device that is running XBMC/Kodi.
+  required: true
+  type: string
+port:
+  description: The HTTP port number.
+  required: false
+  type: integer
+  default: 8080
+tcp_port:
+  description: The TCP port number. Used for websocket connections to Kodi.
+  required: false
+  type: integer
+  default: 9090
+name:
+  description: The name of the device used in the frontend.
+  required: false
+  type: string
+proxy_ssl:
+  description: Connect to kodi with HTTPS and WSS. Useful if Kodi is behind an SSL proxy.
+  required: false
+  type: boolean
+  default: false
+username:
+  description: The XBMC/Kodi HTTP username.
+  required: false
+  type: string
+password:
+  description: The XBMC/Kodi HTTP password.
+  required: false
+  type: string
+turn_on_action:
+  description: Home Assistant script sequence to call when turning on.
+  required: false
+  type: list
+turn_off_action:
+  description: Home Assistant script sequence to call when turning off.
+  required: false
+  type: list
```

Comment:
Not completely sure what type this should be. I think list is the most suitable, since map requires keys to be defined.

Reply:
`SCRIPT_SCHEMA` is a list.

Reply:
Great, thanks! Learning a lot through this.